src/topology/subset_properties.lean

```diff
@@ -1244,6 +1272,106 @@ begin
     { simpa using hs } }
 end
 
+/-- preconnected sets are either contained in, or disjoint to any given clopen set -/
+theorem subset_or_disjoint_of_clopen {α : Type*} [topological_space α] {s t : set α}
+  (h : is_preconnected t) (h1 : is_clopen s) : s ∩ t = ∅ ∨ t ⊆ s :=
+begin
+  apply classical.by_contradiction,
+  intro h2,
+  have h3 : s ∩ t ≠ ∅ ∧ sᶜ ∩ t ≠ ∅,
+  { split,
+    { exact mt or.inl h2 },
+    rw [set.inter_comm, set.ne_empty_iff_nonempty],
+    apply set.inter_compl_nonempty_iff.2,
+    push_neg at h2,
+    exact h2.2 },
+  rw [set.inter_comm, set.ne_empty_iff_nonempty] at h3,
+  conv at h3 in (sᶜ ∩ t ≠ ∅) {rw [set.inter_comm, set.ne_empty_iff_nonempty]},
+  apply set.ne_empty_iff_nonempty.2 (h s sᶜ h1.1 (is_open_compl_iff.2 h1.2) _ h3.1 h3.2),
+  rw [set.inter_compl_self, set.inter_empty],
+  rw [set.union_compl_self],
+  exact set.subset_univ t,
+end
+
+/-- A set `s` is preconnected if and only if
+for every cover by two closed sets that are disjoint on `s`,
+it is contained in one of the two covering sets -/
+theorem is_preconnected_iff_subset_of_disjoint_closed {α : Type*} {s : set α} [topological_space α] :
+  is_preconnected s ↔
+  ∀ (u v : set α) (hu : is_closed u) (hv : is_closed v) (hs : s ⊆ u ∪ v) (huv : s ∩ (u ∩ v) = ∅),
+  s ⊆ u ∨ s ⊆ v :=
+begin
+  split; intro h,
+  { intros u v hu hv hs huv,
+    rw is_preconnected_closed_iff at h,
+    specialize h u v hu hv hs,
+    contrapose! huv,
+    rw set.ne_empty_iff_nonempty,
+    simp [set.not_subset] at huv,
+    rcases huv with ⟨⟨x, hxs, hxu⟩, ⟨y, hys, hyv⟩⟩,
+    have hxv : x ∈ v := or_iff_not_imp_left.mp (hs hxs) hxu,
+    have hyu : y ∈ u := or_iff_not_imp_right.mp (hs hys) hyv,
+    exact h ⟨y, hys, hyu⟩ ⟨x, hxs, hxv⟩ },
+  { rw is_preconnected_closed_iff,
+    intros u v hu hv hs hsu hsv,
+    rw ← set.ne_empty_iff_nonempty,
+    intro H,
+    specialize h u v hu hv hs H,
+    contrapose H,
+    apply set.ne_empty_iff_nonempty.mpr,
+    cases h,
+    { rcases hsv with ⟨x, hxs, hxv⟩, exact ⟨x, hxs, ⟨h hxs, hxv⟩⟩ },
+    { rcases hsu with ⟨x, hxs, hxu⟩, exact ⟨x, hxs, ⟨hxu, h hxs⟩⟩ } }
+end
+
+/-- A closed set `s` is preconnected if and only if
+for every cover by two closed sets that are disjoint,
+it is contained in one of the two covering sets -/
+theorem is_preconnected_iff_subset_of_fully_disjoint_closed {s : set α} (hs : is_closed s) :
+  is_preconnected s ↔
+  ∀ (u v : set α) (hu : is_closed u) (hv : is_closed v) (hss : s ⊆ u ∪ v) (huv : u ∩ v = ∅),
+  s ⊆ u ∨ s ⊆ v :=
+begin
+  split,
+  { intros,
+    apply is_preconnected_iff_subset_of_disjoint_closed.1 ᾰ u v hu hv hss,
+    rw huv,
+    exact set.inter_empty s },
+  intro H,
+  rw is_preconnected_iff_subset_of_disjoint_closed,
+  intros u v hu hv hss huv,
+  have H1 := H (u ∩ s) (v ∩ s),
+  rw [(@set.subset_inter_iff _ u s s), (@set.subset_inter_iff _ v s s)] at H1,
```

Comment:
There are only two places where the rewrite can happen, so you don't need to be specific:
```suggestion
  rw [set.subset_inter_iff, set.subset_inter_iff] at H1,
```
will work as well.